diff --git a/src/assets/CMakeLists.txt b/src/assets/CMakeLists.txt
--- a/src/assets/CMakeLists.txt
+++ b/src/assets/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Copyright (C) 2024 The Qt Company Ltd.
 # SPDX-License-Identifier: BSD-3-Clause
-add_subdirectory(icons)
+#add_subdirectory(icons)
 
 if (NOT INTEGRITY AND TARGET Qt6::Network AND TARGET Qt6::Concurrent)
     add_subdirectory(downloader)
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -62,6 +62,7 @@ qt_internal_add_module(Gui
     FEATURE_DEPENDENCIES
         Qt::Network
     QMAKE_MODULE_CONFIG "${qmake_module_config}"
+    EXCEPTIONS
     SOURCES
         compat/removed_api.cpp
         image/qabstractfileiconengine.cpp image/qabstractfileiconengine_p.h
diff --git a/src/plugins/platforms/windows/CMakeLists.txt b/src/plugins/platforms/windows/CMakeLists.txt
--- a/src/plugins/platforms/windows/CMakeLists.txt
+++ b/src/plugins/platforms/windows/CMakeLists.txt
@@ -9,6 +9,7 @@ qt_internal_add_plugin(QWindowsIntegrationPlugin
     OUTPUT_NAME qwindows
     PLUGIN_TYPE platforms
     DEFAULT_IF "windows" IN_LIST QT_QPA_PLATFORMS
+    EXCEPTIONS
     SOURCES
         main.cpp
         qtwindowsglobal.h
diff --git a/src/widgets/CMakeLists.txt b/src/widgets/CMakeLists.txt
--- a/src/widgets/CMakeLists.txt
+++ b/src/widgets/CMakeLists.txt
@@ -8,6 +8,7 @@
 qt_internal_add_module(Widgets
     QMAKE_MODULE_CONFIG uic
     PLUGIN_TYPES styles
+    EXCEPTIONS
     SOURCES
         compat/removed_api.cpp
         itemviews/qfileiconprovider.cpp itemviews/qfileiconprovider.h itemviews/qfileiconprovider_p.h
